test/lint.test.js

```diff
@@ -0,0 +1,52 @@
+const expect = require("expect");
+
+const lint = require("../lib/lint");
+
+describe("Linter", function() {
+    var mockGrammar, mockOpts, writeSpy;
+
+    beforeEach(function () {
+        mockGrammar = {
+            rules: [],
+            config: {}
+        };
+        mockOpts = {
+            out: {
+                write() {}
+            }
+        };
+        writeSpy = expect.spyOn(mockOpts.out, "write");
+    })
+
+    it("runs without warnings on empty rules", function () {
+        lint(mockGrammar, mockOpts);
+        expect(writeSpy).toNotHaveBeenCalled();
+    });
+
+    it("warns about undefined symbol", function () {
+        mockGrammar.rules = [
+            {name: "a", symbols: ["non-existent"]}
+        ];
+        lint(mockGrammar, mockOpts);
+        expect(writeSpy).toHaveBeenCalled();
+    });
+
+    it("doesn't warn about defined symbol", function () {
+        mockGrammar.rules =  [
+            {name: "a", symbols: []},
+            {name: "b", symbols: ["a"]}
+        ];
+        lint(mockGrammar, mockOpts);
+        expect(writeSpy).toNotHaveBeenCalled();
+    });
+
+    it("doesn't warn about duplicate symbol", function () {
```

Comment:
So, is that as expected? Would be better if the linter warned about duplicate rules right?

Reply:
No, this is definitely expected: we want to allow multiple rules for same non-terminal :-)

Reply:
Oh, you mean _duplicate_ rules, as in, with the same name and symbols. That might be a nice thing to test for, but I think it's outside the scope of this PR :/

Reply:
Well, I only half know what I am doing here. [Lark handles the new multi-include](https://github.com/erezsh/lark/pull/28#issuecomment-334559868) test very differently and complains loudly about re-declared things, is that related? 

How would you change this test  for the actual desired behavior? Add a symbol to the second a rule?

```diff
             {name: "a", symbols: []},
-            {name: "a", symbols: []},
+            {name: "a", symbols: ["b"]},
             {name: "b", symbols: ["a"]}
```

Reply:
@tjvr could you give some feedback on this please

Reply:
I suggest leaving this as-is. The current behaviour is expected.